assembly/src/tokens/stream.rs

```diff
@@ -95,9 +124,10 @@ impl<'a> TokenStream<'a> {
         self.pos
     }
 
-    /// Returns the current lines count for the stream.
-    pub fn num_lines(&self) -> usize {
-        self.lines[self.pos.min(self.lines.len().saturating_sub(1))]
+    /// Returns the [SourceLocation] linked to the current [Token].
+    pub fn location(&self) -> &SourceLocation {
+        let idx = self.pos.min(self.locations.len().saturating_sub(1));
+        &self.locations[idx]
     }
```

Comment:
Do we actually need this method? It seems like we use it only in combination with EOF error - so, maybe it should be something like `last_location()` or maybe `eof_location()` (i.e., last location before EOF).

Reply:
This is just a getter, and can be safely removed without problems

Reply:
renamed on https://github.com/0xPolygonMiden/miden-vm/pull/861/commits/14bedf947eb59c3243de1feb117e3e3d9783c8f7

I think `eof_location` makes more sense because we unambiguously specify this is the EOF of the source